Integration test `SchemaAgreementTest.should_unset_flag_on_failed_agreement` is flaky

Seem that on branch enterprise-2022.2 tests `SchemaAgreementTest.should_unset_flag_on_failed_agreement` keep 
on failing with like:

```
23:57:25  Starting com.datastax.driver.core.SchemaAgreementTest.should_unset_flag_on_failed_agreement [Test #317]...
23:57:25  FAILED : should_unset_flag_on_failed_agreement
23:57:25  Test   : 00:00:00
23:57:25  Elapsed: 01:09:02
23:57:25  
23:57:25      4141961 [cluster5588-reconnection-1] ERROR com.datastax.driver.core.ControlConnection - [Control connection] Cannot connect to any host, scheduling retry in 512000 milliseconds
23:57:25      4141972 [cluster5588-nio-worker-19] WARN  com.datastax.driver.core.Connection - Error creating netty channel to /127.0.1.2:38783
23:57:25  io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.1.2:38783
23:57:25  	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
23:57:25  	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:716)
23:57:25  	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:225)
23:57:25  	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
23:57:25  	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:634)
23:57:25  	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:581)
23:57:25  	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:498)
23:57:25  	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:460)
23:57:25  	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:131)
23:57:25  	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
23:57:25  	at java.lang.Thread.run(Thread.java:750)
23:57:25  Caused by: java.net.ConnectException: Connection refused
23:57:25  	... 11 more
23:57:30      4145646 [main] WARN  com.datastax.driver.core.CCMBridge - Executing: ccm stop --config-dir=/tmp/1667511261457-0
23:57:30      4146268 [main] INFO  com.datastax.driver.core.CCMBridge - Error during tests, kept C* logs in /tmp/1667511261457-0
23:57:30  [ERROR] Tests run: 326, Failures: 1, Errors: 0, Skipped: 12, Time elapsed: 4,148.029 s <<< FAILURE! - in TestSuite
23:57:30  [ERROR] should_unset_flag_on_failed_agreement(com.datastax.driver.core.SchemaAgreementTest)  Time elapsed: 0.01 s  <<< FAILURE!
23:57:30  com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
23:57:30  	at com.datastax.driver.core.SchemaAgreementTest.should_unset_flag_on_failed_agreement(SchemaAgreementTest.java:55)
23:57:30  Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
23:57:30  	at com.datastax.driver.core.SchemaAgreementTest.should_unset_flag_on_failed_agreement(SchemaAgreementTest.java:55)
23:57:30  
23:57:30      4146330 [Thread-76] WARN  com.datastax.driver.core.CCMBridge - Executing: ccm stop --config-dir=/tmp/1667511790109-0
23:57:30      4146331 [Thread-80] WARN  com.datastax.driver.core.CCMBridge - Executing: ccm stop --config-dir=/tmp/1667512049589-0
23:57:33      4149960 [Thread-80] WARN  com.datastax.driver.core.CCMBridge - Executing: ccm remove --config-dir=/tmp/1667512049589-0
23:57:33      4149965 [Thread-76] WARN  com.datastax.driver.core.CCMBridge - Executing: ccm remove --config-dir=/tmp/1667511790109-0
23:57:36  [INFO] 
23:57:36  [INFO] Results:
23:57:36  [INFO] 
23:57:36  [ERROR] Failures: 
23:57:36  [ERROR]   SchemaAgreementTest.should_unset_flag_on_failed_agreement:55 ? NoHostAvailable
23:57:36  [INFO] 
23:57:36  [ERROR] Tests run: 323, Failures: 1, Errors: 0, Skipped: 9
```

https://jenkins.scylladb.com/job/enterprise-2022.2/job/driver-tests/job/java-driver-matrix-test/8/console
https://jenkins.scylladb.com/job/enterprise-2022.2/job/driver-tests/job/java-driver-matrix-test/7/console
https://jenkins.scylladb.com/job/enterprise-2022.2/job/driver-tests/job/java-driver-matrix-test/6/console


seems like cluster with node 127.0.1.2:38783, is being used in some tests before, and it's not in a health state

this test seems to be working in master and 5.1 just fine:
https://jenkins.scylladb.com/job/scylla-master/job/driver-tests/job/java-driver-matrix-test/894/testReport/3.7.1-scylla-2.com.datastax.driver.core/SchemaAgreementTest/should_unset_flag_on_failed_agreement/history/
 https://jenkins.scylladb.com/view/scylla-5.1/job/scylla-5.1/job/driver-tests/job/java-driver-matrix-test/7/console

--- a/driver-core/src/test/java/com/datastax/driver/core/ControlConnectionTest.java
+++ b/driver-core/src/test/java/com/datastax/driver/core/ControlConnectionTest.java
@@ -68,7 +68,7 @@ public class ControlConnectionTest extends CCMTestsSupport {
   static final Logger logger = LoggerFactory.getLogger(ControlConnectionTest.class);
 
   @Test(groups = "short")
-  @CCMConfig(numberOfNodes = 2)
+  @CCMConfig(numberOfNodes = 2, dirtiesContext = true)
   public void should_prevent_simultaneous_reconnection_attempts() throws InterruptedException {
 
     // Custom load balancing policy that counts the number of calls to newQueryPlan().

--- a/driver-core/src/test/java/com/datastax/driver/core/SchemaAgreementTest.java
+++ b/driver-core/src/test/java/com/datastax/driver/core/SchemaAgreementTest.java
@@ -47,7 +47,7 @@ public class SchemaAgreementTest extends CCMTestsSupport {
     assertThat(rs.getExecutionInfo().isSchemaInAgreement()).isTrue();
   }
 
-  @Test(groups = "short", priority = 1)
+  @Test(groups = "short")
   public void should_unset_flag_on_failed_agreement() {
     // Setting to 0 results in no query being set, so agreement fails
     ProtocolOptions protocolOptions = cluster().getConfiguration().getProtocolOptions();

--- a/driver-core/src/test/java/com/datastax/driver/core/SchemaAgreementTest.java
+++ b/driver-core/src/test/java/com/datastax/driver/core/SchemaAgreementTest.java
@@ -54,6 +54,13 @@ public class SchemaAgreementTest extends CCMTestsSupport {
     protocolOptions.maxSchemaAgreementWaitSeconds = 0;
     ResultSet rs = session().execute(String.format(CREATE_TABLE, COUNTER.getAndIncrement()));
     assertThat(rs.getExecutionInfo().isSchemaInAgreement()).isFalse();
+
+    // Execute a dummy query, that will wait for schema agreement,
+    // so that the next tests that run afterwards have a clean state.
+    protocolOptions = cluster().getConfiguration().getProtocolOptions();
+    protocolOptions.maxSchemaAgreementWaitSeconds = 10;
+    rs = session().execute(String.format(CREATE_TABLE, COUNTER.getAndIncrement()));
+    assertThat(rs.getExecutionInfo().isSchemaInAgreement()).isTrue();
   }
 
   @Test(groups = "short")

